libs/api/repository/src/lib/gn4/gn4-repository.ts

```diff
@@ -137,6 +141,27 @@ export class Gn4Repository implements RecordsRepositoryInterface {
       )
   }
 
+  getFeatureCatalog(
+    metadataUuid: string,
+    approvedVersion?: boolean
+  ): Observable<DatasetFeatureCatalog | null> {
+    return this.gn4RecordsApi
+      .getFeatureCatalog(metadataUuid, approvedVersion)
+      .pipe(
+        map((results: FeatureResponseApiModel) => {
+          if (results.decodeMap) {
+            const features = Object.keys(results.decodeMap).map((key) => {
+              const feature = results.decodeMap[key]
+              return { name: feature[0], title: feature[1] }
+            })
+            return { features } as DatasetFeatureCatalog
+          }
+          return null
```

Comment:
```suggestion
          if (!results.decodeMap) {
            return null;
          }
          const features = Object.keys(results.decodeMap).map((key) => {
            const feature = results.decodeMap[key]
            return { name: feature[0], title: feature[1] }
          })
          return { features } as DatasetFeatureCatalog
```

This way we're keeping the "happy path" on the left :)